CMakeLists.txt

```diff
@@ -2,5 +2,47 @@ cmake_minimum_required(VERSION 3.10)
 
 project(vesuvius C)
 
-add_executable(vesuvius_example example2.c)
-add_executable(vesuvius_tests runtests.c)
\ No newline at end of file
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+add_executable(vesuvius_example example.c)
+add_executable(vesuvius_example2 example2.c)
+add_executable(vesuvius_tests runtests.c)
+
+include_directories(third-party/json.h)
+
+find_package(Blosc2 REQUIRED)
+find_package(CURL REQUIRED)
+
+add_compile_options(-g3 -Wall -Wextra)
+if(UNIX)
+    add_link_options(-rdynamic)
```

Comment:
Is this actually necessary? I'm slightly surprised but it's been awhile.

Reply:
are you referring to the `-rdynamic` link option? It's not necessary for compiling, but it helps with the `ASSERT` macro for printing the backtrace with function symbols rather than just some virtual address. E.g.


    chunk must be less than 1 gb
    Assertion failed!
    Expression: chunk_dims[0] * chunk_dims[1]*chunk_dims[2]*4 < 1024*1024*1
    Location  : /home/forrest/volcano/third-party/vesuvius-c/vesuvius-c.h:4405
    Function  : vs_vol_get_chunk
    
    Backtrace:
      [0] /home/forrest/volcano/cmake-build-debug/volcano(+0xd219) [0x5628c64f7219]
      [1] /home/forrest/volcano/cmake-build-debug/volcano(+0xd3e6) [0x5628c64f73e6]
      [2] /home/forrest/volcano/cmake-build-debug/volcano(vs_vol_get_chunk+0xb4) [0x5628c6500f4f]
      [3] /home/forrest/volcano/cmake-build-debug/volcano(main+0x80) [0x5628c650583b]
      [4] /lib/x86_64-linux-gnu/libc.so.6(+0x2a1ca) [0x7f76531711ca]
      [5] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x8b) [0x7f765317128b]
      [6] /home/forrest/volcano/cmake-build-debug/volcano(_start+0x25) [0x5628c64ee775]
 
You wouldn't get the `vs_vol_get_chunk` there without `-rdynamic`. Though I need to change this to use `target_link_options` because `add_link_options` _just doesn't work_ and I don't know why

Reply:
OK, add a comment then. I suspect add_link_options vs target_link_options is just one of those cmake things. *shrug*

Reply:
addressed